src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -102,13 +108,19 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
   QMenu *menu = new QMenu( this );
   // common angles
   QActionGroup *angleButtonGroup = new QActionGroup( menu ); // actions are exclusive for common angles
-  mCommonAngleActions = QMap<QAction *, int>();
-  QList< QPair< int, QString > > commonAngles;
-  commonAngles << QPair<int, QString>( 0, tr( "Do Not Snap to Common Angles" ) );
-  commonAngles << QPair<int, QString>( 30, tr( "Snap to 30° Angles" ) );
-  commonAngles << QPair<int, QString>( 45, tr( "Snap to 45° Angles" ) );
-  commonAngles << QPair<int, QString>( 90, tr( "Snap to 90° Angles" ) );
-  for ( QList< QPair< int, QString > >::const_iterator it = commonAngles.constBegin(); it != commonAngles.constEnd(); ++it )
+  mCommonAngleActions = QMap<QAction *, double>();
+  QList< QPair< double, QString > > commonAngles;
+  QString menuText;
+  QList<double> anglesDouble( { 0.0, 5.0, 10.0, 15.0, 18.0, 22.5, 30.0, 45.0, 90.0} );
+  for ( QList<double>::const_iterator it = anglesDouble.constBegin(); it != anglesDouble.constEnd(); ++it )
+  {
+    if ( *it == 0 )
+      menuText = tr( "Do not snap to common angles" );
+    else
+      menuText = QString( tr( "%1, %2, %3, %4°..." ) ).arg( *it, 0, 'f', 1 ).arg( *it * 2, 0, 'f', 1 ).arg( *it * 3, 0, 'f', 1 ).arg( *it * 4, 0, 'f', 1 );
```

Comment:
'...' -> '…' (proper ellipsis character instead of three dots)

Reply:
Done. Thank you.